src/trio/_threads.py

```diff
@@ -369,29 +361,30 @@ def worker_fn() -> RetT:
             del PARENT_TASK_DATA.task_register
 
     context = contextvars.copy_context()
-    # Partial confuses type checkers, coerce to a callable.
-    contextvars_aware_worker_fn: Callable[[], RetT] = functools.partial(context.run, worker_fn)  # type: ignore[assignment]
+    # Trio doesn't use current_async_library_cvar, but if someone
+    # else set it, it would now shine through since
+    # sniffio.thread_local isn't set in the new thread. Make sure
+    # the new thread sees that it's not running in async context.
+    context.run(current_async_library_cvar.set, None)
```

Comment:
I thought we didn't use this anymore. As of https://github.com/python-trio/trio/pull/2700

Reply:
Ohh sorry I should have read the comment better. Ack.